pkg/awsutils/awsutils.go

```diff
@@ -1225,7 +1288,13 @@ func (cache *EC2InstanceMetadataCache) GetVPCIPv4CIDR() string {
 
 // GetVPCIPv4CIDRs returns VPC CIDRs
 func (cache *EC2InstanceMetadataCache) GetVPCIPv4CIDRs() []*string {
-	return cache.vpcIPv4CIDRs
+	var vpcIPv4CIDRs []*string
+	cache.vpcIPv4CIDRs.RLock()
+	defer cache.vpcIPv4CIDRs.RUnlock()
+	for vpcIPv4CIDR := range cache.vpcIPv4CIDRs.data{
+		vpcIPv4CIDRs = append(vpcIPv4CIDRs, aws.String(vpcIPv4CIDR))
```

Comment:
I have a sense that this doesn't work. I suspect it will return the last CIDR in the set every time since we return the address of `vpcIPv4CIDR` and that value will be updated on each loop. Similar to [this code](https://github.com/aws/amazon-vpc-cni-k8s/blob/9c863340dac6fc959636c57ae4a1706d02748874/pkg/awsutils/awsutils.go#L500-L506).

Reply:
Thanks for catching this 🙏